packages/ui-react/src/components/Sidebar/Sidebar.tsx

```diff
@@ -18,16 +19,62 @@ const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, children }) => {
   const { t } = useTranslation('menu');
   const lastFocusedElementRef = useRef<HTMLElement | null>(null);
   const sidebarRef = useRef<HTMLDivElement>(null);
+  const [visible, setVisible] = useState(false);
+
   const htmlAttributes = { inert: !isOpen ? '' : undefined }; // inert is not yet officially supported in react. see: https://github.com/facebook/react/pull/24730
 
   useEffect(() => {
     if (isOpen) {
+      // Before inert on the body is applied in Layout, we need to set this ref
       lastFocusedElementRef.current = document.activeElement as HTMLElement;
+
+      // When opened, adjust the margin-right to accommodate for the scrollbar width to prevent UI shifts in background
+      document.body.style.marginRight = `${scrollbarSize()}px`;
```

Comment:
The `scrollbarSize` returns the actual scrollbar width of the current browser. When opening a modal (or sidebar), the scrollbar will temporarily hide to prevent background scrolling. But when the scrollbar is visible, hiding it will cause a layout shift.

This `marginRight` prevents the layout shift by reserving the previous scrollbar spacing.

This is the same method used in the Modal component. We have a ticket to look into this further in the next sprint. Possibly also introducing some modal manager because stacking multiple modals doesn't work very well at the moment.

Reply:
Makes sense to me. Though I didn't manage to get any value except 0. Can be also connected with the way Chrome / Safari  handle `overflow: scroll` on Mac now.